src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -150,12 +150,28 @@ public override void DeleteFile(string fullPath)
             if (Interop.Sys.Unlink(fullPath) < 0)
             {
                 Interop.ErrorInfo errorInfo = Interop.Sys.GetLastErrorInfo();
-                // ENOENT means it already doesn't exist; nop
-                if (errorInfo.Error != Interop.Error.ENOENT)
+                switch(errorInfo.Error)
                 {
-                    if (errorInfo.Error == Interop.Error.EISDIR)
+                    case Interop.Error.ENOENT:
+                        // ENOENT means it already doesn't exist; nop
+                        return;
+                    case Interop.Error.EROFS:
+                        // EROFS means the file system is read-only
+                        // Need to manually check file existence
+                        // github.com/dotnet/corefx/issues/21273
```

Comment:
Generally I would rely on commit descriptions to find old issues if we need them. Otherwise the tests get littered with old bug ID's. In this case I think it's sufficient to simply explain that this makes it consistent with Mono, or event say nothing at all since it's fairly obvious.